pkg/utils/termination/suite_test.go

```diff
@@ -0,0 +1,110 @@
+/*
```

Comment:
Test Scenarios:
1. Call Delete only once (which is the if the status condition is already set, then don't call Delete again)
2. Call Delete and return NotFound
3. Call Delete and then Get and return NotFound
4. Handle not having a providerID
5.